### DISCUSSION TITLE
[ 命名建議 ] LeetCode 141. Linked List Cycle

### DISCUSSION THREAD

**Discussion by LPenny-github** [EngTW/English-for-Programmers#121] [Q&A] [answered]
LeetCode:
https://leetcode.com/problems/linked-list-cycle/

線上教材:
https://blog.techbridge.cc/2019/11/22/leetcode-pattern-fast-and-slow-pointer/

---

再麻煩 @twy30 給予命名建議 orz 感激不盡

```csharp
public class Solution
{
    public bool HasCycle(ListNode head)
    {
        ListNode headNode = head;
        ListNode fastNodeDetector = head;
        ListNode slowNodeDetector = head;
        while (fastNodeDetector?.next != null)
        {
            fastNodeDetector = fastNodeDetector.next.next;
            slowNodeDetector = slowNodeDetector.next;

            if (fastNodeDetector == slowNodeDetector)
            {
                return true;
            }
        }
        return false;
    }
}
```

**Comment by twy30** [comment, marked as answer]
@LPenny-github 

你好 😊

我覺得這題用 `fast`, `slow` 很棒，直覺好懂 👍

我會把 `NodeDetector` 換成 `Iterator` ( https://zh.wikipedia.org/zh-tw/%E8%BF%AD%E4%BB%A3%E5%99%A8 ), 變成

* `fastIterator`
* `slowIterator`

這兩種寫法在語意上有些差異：

* `fastNodeDetector`
  * 快 的 節點偵測器
  * 快節點 的 偵測器
* `fastIterator`
  * 快 的 Iterator (疊代器)

可以參考看看 😊

**Reply by LPenny-github** [reply]
> 我覺得這題用 fast, slow 很棒，直覺好懂 👍

是參考教材的用法，我還沒有獨創的能力 😅

謝謝 @twy30 大大的指點 orz
